docs/source/kpca.rst

```diff
@@ -0,0 +1,121 @@
+
+.. function:: principalvars(M)
+
+    The variances of principal components.
+
+Transformation and Construction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The package provides methods to do so:
+
+.. function:: transform(M, x)
+
+    Transform observations ``x`` into principal components.
+
+    Here, ``x`` can be either a vector of length ``d`` or a matrix where each column is an observation.
+
+.. function:: reconstruct(M, y)
+
+    Approximately reconstruct observations from the principal components given in ``y``.
+
+    Here, ``y`` can be either a vector of length ``p`` or a matrix where each column gives the principal components for an observation.
+
+
+Data Analysis
+~~~~~~~~~~~~~~~
+
+One can use the ``fit`` method to perform kernel PCA over a given dataset.
+
+.. function:: fit(KernelPCA, X; ...)
+
+    Perform kernel PCA over the data given in a matrix ``X``. Each column of ``X`` is an observation.
+
+    This method returns an instance of ``KernelPCA``.
+
+    **Keyword arguments:**
+
+    Let ``(d, n) = size(X)`` be respectively the input dimension and the number of observations:
+
+    =========== =============================================================== ===============
+      name         description                                                   default
+    =========== =============================================================== ===============
+     kernel     The kernel function:                                             ``(x,y)->x'y``
+
+                This functions accepts two vector arguments ``x`` and ``y``,
+                and returns a scalar value.
+    ----------- --------------------------------------------------------------- ---------------
+     solver     The choice of solver:                                            ``:eig``
+
+                - ``:eig``: uses ``eigfact``
+                - ``:eigs``: uses ``eigs`` (always used for sparse data)
+    ----------- --------------------------------------------------------------- ---------------
+     maxoutdim  Maximum output dimension.                                        ``min(d, n)``
+    ----------- --------------------------------------------------------------- ---------------
+     inverse    Performs calculation for inverse transform for non-precomputed   ``false``
+                kernels.
+    ----------- --------------------------------------------------------------- ---------------
+     β          Hyperparameter of the ridge regression that learns the           ``1.0``
+                inverse transform (when ``inverse`` is ``true``).
+    ----------- --------------------------------------------------------------- ---------------
+     etol        Convergence tolerance for ``eigs`` solver                       ``0.0``
+    ----------- --------------------------------------------------------------- ---------------
+     etot        Maximum number of iterations for ``eigs`` solver                ``300``
```

Comment:
The name isn't very clear. Maybe `maxiter` (as for `eigs`)? BTW, the "e" in `etol` may not be needed either (and in the future it could used with other solvers).